File: src/sandwich_buttons.rb
class SandwichButtons
  attr_reader :buttons, :buttons_html

  def initialize
    @buttons = ["<button onclick=\"document.body.background = ''\">Sandwich Off</button>", "<button onclick=\"document.body.background = 'sandwich.jpg'\">Sandwich On</button>\n"]
    @buttons_html = @buttons.join
  end
end

File: elixir_onepg/lib/onepg_ex/sandwich_buttons.ex
defmodule OnepgEx.SandwichButtons do
  @moduledoc """
  Buttons for turning the sandwich background on and off. Essential.
  """
  @button_nodes [
    "<button onclick=\"document.body.background = ''\">Sandwich Off</button>",
    "<button onclick=\"document.body.background = 'sandwich.jpg'\">Sandwich On</button>\n"
  ]

  defstruct button_nodes: @button_nodes,
            html_document: %Onepg.HtmlDocument{
              nodes: @button_nodes,
              body: Enum.join(@button_nodes, "\n")
            }

  def buttons do
    %__MODULE__{}
  end
end
